pysph/solver/tools.py

```diff
@@ -81,9 +81,78 @@ def __init__(self, app, array_name, props, freq=100, xi=None, yi=None,
         )
 
     def post_step(self, solver):
-        if solver.count%self.freq == 0 and solver.count > 0:
+        if solver.count % self.freq == 0 and solver.count > 0:
             self.interp.nnps.update()
             data = dict(x=self.xi, y=self.yi, z=self.zi)
             for prop in self.props:
                 data[prop] = self.interp.interpolate(prop)
             self.array.set(**data)
+
+
+class DensityCorrection(Tool):
+    """
+    A tool to reinitialize the density of the fluid particles
+    """
+
+    def __init__(self, app, arr_names, corr='shepard', freq=10, kernel=None):
+        """
+        Parameters
+        ----------
+
+        app : pysph.solver.application.Application.
+            The application instance.
+        arr_names : array
+            Names of the particle arrays whose densities needs to be
+            reinitialized.
+        corr : str
+            Name of the density reinitialization operation
+        freq : int
+            Frequency of reinitialization.
+        kernel: any kernel from pysph.base.kernels
+
+        """
+        self.freq = freq
+        self.corr = corr
```

Comment:
Add an `assert corr in ['shepard']` with a suitable error message so users get an error with an invalid corr.